sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosTracerTest.java

```diff
@@ -0,0 +1,375 @@
+            cosmosAsyncDatabase.getId(), 15);
+
+
+        cosmosAsyncContainer.getScripts().getStoredProcedure(procedureProperties.getId()).replace(resultSproc).block();
+        Mockito.verify(tracerProvider, Mockito.times(16)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "replaceStoredProcedure." + cosmosAsyncContainer.getId(), context,
+            cosmosAsyncDatabase.getId(), 16);
+
+        cosmosAsyncContainer.getScripts().readAllStoredProcedures(new CosmosQueryRequestOptions()).byPage().single().block();
+
+        cosmosAsyncContainer.getScripts().getStoredProcedure(procedureProperties.getId()).delete().block();
+        Mockito.verify(tracerProvider, Mockito.times(18)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "deleteStoredProcedure." + cosmosAsyncContainer.getId(), context,
+            cosmosAsyncDatabase.getId(), 18);
+    }
+
+    @AfterClass(groups = {"emulator"}, timeOut = SETUP_TIMEOUT)
+    public void afterClass() {
+        LifeCycleUtils.closeQuietly(client);
+    }
+
+    private static CosmosUserDefinedFunctionProperties getCosmosUserDefinedFunctionProperties() {
+        CosmosUserDefinedFunctionProperties udf =
+            new CosmosUserDefinedFunctionProperties(UUID.randomUUID().toString(), "function() {var x = 10;}");
+        return udf;
+    }
+
+    private static CosmosTriggerProperties getCosmosTriggerProperties() {
+        CosmosTriggerProperties trigger = new CosmosTriggerProperties(UUID.randomUUID().toString(), "function() {var " +
+            "x = 10;}");
+        trigger.setTriggerOperation(TriggerOperation.CREATE);
+        trigger.setTriggerType(TriggerType.PRE);
+        return trigger;
+    }
+
+    private static CosmosStoredProcedureProperties getCosmosStoredProcedureProperties() {
+        CosmosStoredProcedureProperties storedProcedureDef =
+            new CosmosStoredProcedureProperties(UUID.randomUUID().toString(), "function() {var x = 10;}");
+        return storedProcedureDef;
+    }
+
+    private List<Tracer> getMockTracer(Tracer tracer) {
+        List<Tracer> tracerList = new ArrayList<>();
+        tracerList.add(tracer);
+        Mockito.when(tracer.start(Matchers.anyString(), Matchers.any(Context.class))).thenReturn(Context.NONE);
+        return tracerList;
+    }
+
+    private void verifyTracerAttributes(Tracer mockTracer, String methodName, Context context, String databaseName,
+                                        int numberOfTimesCalledWithinTest) {
+        if (databaseName != null) {
+            Mockito.verify(mockTracer, Mockito.times(numberOfTimesCalledWithinTest)).setAttribute(TracerProvider.DB_INSTANCE,
+                databaseName, context);
```

Comment:
What is happening here, why are we setting attributes in a method that's verifying tracer attributes?

Reply:
This   will not set any attribute,it is just mockito verify.